### PR TITLE
Sort stealthKeyLogs

### PR DESCRIPTION
**Pull request by garyghayrat**

### CODE REVIEW

File: umbra-js/src/utils/utils.ts [hunk 1]
```diff
@@ -245,7 +245,12 @@ export async function getBlockNumberUserRegistered(address: string, provider: St
   address = getAddress(address); // address input validation
   const registry = new StealthKeyRegistry(provider);
   const filter = registry._registry.filters.StealthKeyChanged(address, null, null, null, null);
-  const stealthKeyLogs = await registry._registry.queryFilter(filter);
+  let stealthKeyLogs = await registry._registry.queryFilter(filter);
+  if (stealthKeyLogs.length > 1) {
+    stealthKeyLogs = stealthKeyLogs.sort(function (a, b) {
```
Comment (mds1) [inline umbra-js/src/utils/utils.ts:250]: We should add a test for this method, or if we don't want the RPC request during tests, pull the sorting into a separate method and unit test that

Comment (apbendi) [reply umbra-js/src/utils/utils.ts:250]: I would pull the sort method out into something that can be unit tested, then I would call sort regardless of the length of the logs. It should be extremely fast and it's not worth short circuiting.

Comment (garyghayrat) [reply umbra-js/src/utils/utils.ts]: Added test https://github.com/ScopeLift/umbra-protocol/pull/567/commits/b70243b6b34b90fa1aa8ac95ddbf26e1affa8902
Removed `if` checking https://github.com/ScopeLift/umbra-protocol/pull/567/commits/b3957ac615b891271ae6520ef630aeab3aa9a730

File: umbra-js/src/utils/utils.ts [hunk 2]
```diff
@@ -245,11 +246,19 @@ export async function getBlockNumberUserRegistered(address: string, provider: St
   address = getAddress(address); // address input validation
   const registry = new StealthKeyRegistry(provider);
   const filter = registry._registry.filters.StealthKeyChanged(address, null, null, null, null);
-  const stealthKeyLogs = await registry._registry.queryFilter(filter);
+  let stealthKeyLogs = await registry._registry.queryFilter(filter);
+  stealthKeyLogs = sortStealthKeyLogs(stealthKeyLogs);
```
[the remark this answers is not in the document]
Comment (mds1) [reply umbra-js/src/utils/utils.ts]: Looks like you added a commit that resolves this as I was writing it up, so resolving!

File: umbra-js/src/utils/utils.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ import { default as Resolution } from '@unstoppabledomains/resolution';
 import { StealthKeyRegistry } from '../classes/StealthKeyRegistry';
 import { TxHistoryProvider } from '../classes/TxHistoryProvider';
 import { EthersProvider, TransactionResponseExtended } from '../types';
+import { Event } from 'ethers';
```
[the remark this answers is not in the document]
Comment (garyghayrat) [reply umbra-js/src/utils/utils.ts]: https://github.com/ScopeLift/umbra-protocol/pull/567/commits/361b461493a37a1c385aa439dd8f7d303579189f Updated it in tests as well

**Review by apbendi — APPROVED** [review]
Code looks good, tests pass, and frontend works with an account where I set stealth keys twice with sends in between. Nice work @garyghayrat.

**Review by alexkeating — APPROVED** [review]

**Pull request merged**